cmd/timestamp-cli/app/verify.go

```diff
@@ -85,11 +93,148 @@ func runVerify() (interface{}, error) {
 		return nil, err
 	}
 
-	err = verification.VerifyTimestampResponse(tsrBytes, artifact, certPool)
+	opts, err := newVerifyOpts()
+	if err != nil {
+		return verifyCmdOutput{TimestampPath: tsrPath}, fmt.Errorf("failed to created VerifyOpts: %w", err)
+	}
+
+	err = verification.VerifyTimestampResponse(tsrBytes, artifact, certPool, opts)
 
 	return &verifyCmdOutput{TimestampPath: tsrPath}, err
 }
 
+func newVerifyOpts() (verification.VerifyOpts, error) {
+	opts := verification.VerifyOpts{}
+
+	oid, err := getOid()
+	if err != nil {
+		return verification.VerifyOpts{}, fmt.Errorf("failed to parse value from oid flag: %w", err)
+	}
+	if oid != nil {
+		opts.Oid = oid
+	}
+
+	certPathFlagVal := viper.GetString("cert")
+	if certPathFlagVal != "" {
+		cert, err := createCertFromPEMFile(certPathFlagVal)
+		if err != nil {
+			return verification.VerifyOpts{}, fmt.Errorf("failed to parse cert flag value from PEM file: %w", err)
+		}
+		opts.TsaCertificate = cert
+	}
+
+	roots, intermediates, err := getCerts()
+	if err != nil {
+		return verification.VerifyOpts{}, fmt.Errorf("failed to parse root and intermediate certs from cert-chain flag: %w", err)
+	}
+	if roots != nil && intermediates != nil {
```

Comment:
Suggestion: The code looks good. I just share a suggestion here, feel free to adjust it. 
To avoid all these != nil if statements for all the VerifyOpts fields, you could create functions for verifyOpts as such:

`(v *VerifyOpts) OID(...)` where you set the OID if passed.
`(v *VerifyOpts) Nonce(...)` where you set the Nonce if passed. 
...
So you move all that logic inside these functions and only check for errors here.

Reply:
Since the `VerifyOpts` fields will be initialized to their respective zero values when the struct instance is created, we could just set the values returned by the getter functions like `getCerts` without checking whether the values are the zero values or not. If the values are equivalent to the zero values, nothing will change. But I'm happy to add methods if you prefer.